beacon_node/execution_layer/src/lib.rs

```diff
@@ -239,6 +243,51 @@ impl ExecutionLayer {
         self.engines().any_synced().await
     }
 
+    /// Updates the proposer preparation data provided by validators
+    pub fn update_proposer_preparation_blocking(
+        &self,
+        preparation_data: &[ProposerPreparationData],
+    ) -> Result<(), Error> {
+        self.block_on_generic(|_| async move {
+            self.update_proposer_preparation(preparation_data).await
+        })
+        .unwrap()
```

Comment:
The general ethos (with exceptions) in Lighthouse is "never panic", so we need to gracefully handle this `Result`, otherwise this function might panic during shutdown.

Reply:
Thanks for reviewing properly 👍
Thats the result of not understanding what is going on completely, yet. 😅
I think I got it now and will recheck the rest of my changes for similar issues 😊

For here I think it's best to propagate the error up to the calling function via `?`.
The [calling function](https://github.com/pk910/lighthouse/blob/96d5cf9c0b75d603e7f28f0c9eb127be7ae5b749/beacon_node/http_api/src/lib.rs#L2209-L2215) already handles errors properly and rejects the api call with a `custom_bad_request` in case of an error.
Is that correct?